java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -1568,36 +1569,63 @@ private boolean endsWithReturn(StatementTree stmt) {
   /** removes enum in annotation and the comma if there is more than one enum in it */
   private SuggestedFix.Builder deleteExprWithComma(
       VisitorState state,
-      Tree tree,
       ExpressionTree expressionTree,
       ImmutableSet<ResolvedTestAnnotation> resolvedTestAnnotations,
       SuggestedFix.Builder fixBuilder) {
     for (ResolvedTestAnnotation resolvedTestAnnotation : resolvedTestAnnotations) {
-      if (resolvedTestAnnotation.getFlags().size() > 1) {
-        // the index is used to know if we have to remove the comma after or before the enum
-        int index =
-            IntStream.range(0, resolvedTestAnnotation.getFlags().size())
-                .filter(i -> resolvedTestAnnotation.getFlags().get(i).getValue().equals(xpFlagName))
-                .findFirst()
-                .orElse(-1);
-        if (index != -1) {
-          String replacementString = state.getSourceForNode(tree);
-          if (index == resolvedTestAnnotation.getFlags().size() - 1) {
-            replacementString =
-                replacementString.replace(", " + state.getSourceForNode(expressionTree), "");
-          } else {
-            replacementString =
-                replacementString.replace(state.getSourceForNode(expressionTree) + ", ", "");
-          }
-          fixBuilder.replace(tree, replacementString);
-        }
-      } else {
-        fixBuilder.delete(expressionTree);
+
+      // remove the comma before or after the enum depending on the index
+      int index =
+          IntStream.range(0, resolvedTestAnnotation.getFlags().size())
+              .filter(i -> resolvedTestAnnotation.getFlags().get(i).getValue().equals(xpFlagName))
+              .findFirst()
+              .orElse(-1);
+      if (index != -1) {
+        JCTree node = (JCTree) resolvedTestAnnotation.getSourceTree();
+        int startAbsolute = node.getStartPosition();
+        int lower = ((JCTree) expressionTree).getStartPosition() - startAbsolute;
+        int upper = state.getEndPosition(expressionTree) - startAbsolute;
+        CharSequence source = state.getSourceForNode(node);
+
+        lower = getLower(source, lower, index, resolvedTestAnnotation.getFlags().size());
+
+        upper = getUpper(source, upper);
+
+        fixBuilder.replace(startAbsolute + lower, startAbsolute + upper, "");
       }
     }
     return fixBuilder;
   }
 
+  private int getLower(CharSequence source, int lower, int index, int flagSize) {
+    while (lower >= 0
+        && source.charAt(lower) != '{'
```

Comment:
In what scenarios would we encounter a `{` or a `=` in an Enum List?  Can you please add a test scenario for this?

Reply:
for `=` i was thinking in `@ToggleTesting(treated = TestExperimentName.STALE_FLAG)` but you are right that it is not needed at this stage so i'll remove it
for `{` the scenario is tested in `testRemoveFirstEnumFromAnnotationRemovingCommaAndKeepingOtherEnum` where the enum is at the beginning of the list: `@ToggleTesting(treated = {TestExperimentName.STALE_FLAG, TestExperimentName.OTHER_FLAG})`